Cython/Compiler/Options.py

```diff
@@ -187,6 +187,8 @@ def copy_inherited_directives(outer_directives, **new_directives):
     'auto_pickle': None,
     'cdivision': False,  # was True before 0.12
     'cdivision_warnings': False,
+    'cpow': None,  # was True before 3.0
+    # None (not set by user) is treated as slightly different from False
```

Comment:
Are the differences (between False and None) observable from a user PoV? If so, the description in `docs/src/userguide/source_files_and_compilation.rst` should mention that.

Reply:
They probably are. The difference is that `None` second-guesses a few cases where the user probably did intend the old behaviour:

```
cdef double z = x**y
cdef_function_taking_a_double(x**y)
```

It issues a warning when it does this though. It isn't possible for the user to manually set it to None though, so they can't choose this behaviour.

I'd originally chosen not to document this - it was fairly fiddly to explain anyway and I don't think we really want people coding for this behaviour, but I can reconsider.